libs/src/services/contracts/ContractClient.js

```diff
@@ -0,0 +1,76 @@
+const CONTRACT_INITIALIZING_INTERVAL = 100
+const CONTRACT_INITIALIZING_TIMEOUT = 10000
+
+/*
+ * Base class for instantiating contracts.
+ * Performs a single init of the eth contract the first
+ * time a method on the contract is invoked.
+ */
+class ContractClient {
+  constructor (web3Manager, contractABI, contractRegistryKey, getRegistryAddress) {
+    this.web3Manager = web3Manager
+    this.web3 = web3Manager.getWeb3()
+    this.contractABI = contractABI
```

Comment:
per offline discussion, generate list of known method names on init and cross-reference when using getMethod